provider/machinepool/machine_pool_resource.go

```diff
@@ -778,6 +789,25 @@ func (r *MachinePoolResource) populateState(object *cmv1.MachinePool, state *Mac
 	} else {
 		state.Labels = types.MapNull(types.StringType)
 	}
+
+	// Due to RequiresReplace(), we need to ensure these fields always have a
+	// value, even if it's empty. It will be empty if the cluster is multi-AZ or
+	// if the other (AZ/subnet) value is set. We don't need to set
+	// MultiAvailibilityZone here, because it's set in the validation function
+	// during create.
+	azs := object.AvailabilityZones()
+	if len(azs) == 1 {
+		state.AvailabilityZone = types.StringValue(azs[0])
+	} else {
+		state.AvailabilityZone = types.StringValue("")
```

Comment:
What about setting a default string in the attribute scop also?

Reply:
TIL... defaults. :smile:

Trying to set defaults is causing problems... specifically, it treats them like the user specified the value (i.e., we can't set a different one later in processing). For the single AZ cases where we do need to assign it, we can't.

The docs also say that defaults are applied after validation, so they don't really simplify much there either. I looked at assigning default False to the optional bools (autoscaling, use_spot), but it didn't end up simplifying the runtime validation logic.

Reply:
OK I understand :) 
thank you for your clarification